Feature/subseasonal DCOMROOT removal

ShannonShields-NOAA opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
Testing has been conducted on the command line and proven that $DCOMROOT does not need to be manually set in j-jobs.
- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
No testing is needed; only "export DCOMROOT=${DCOMROOT:-/lfs/h1/ops/$envir/dcom}" has been removed from all j-jobs.
This was done because Steven Earle said "ROOT variables should never be set in production, they get inherited from prod_envir."
- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**
Yes
- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**
Yes
- [ ] Please complete this pull request review by **[Fill in date]**.</br>
ASAP as this should be really quick to check commits.
## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

ShannonShields-NOAA committed:
Remove DCOMROOT path in j-jobs

ShannonShields-NOAA committed:
Remove DCOMROOT path in j-jobs

ShannonShields-NOAA committed:
Remove DCOMROOT path in j-jobs

AliciaBentley-NOAA commented:
@ShannonShields-NOAA Remove line 49 that refers to “www” path in: [parm/evs_config/subseasonal/config.evs.subseasonal.grid2grid.plots.sea_ice](https://github.com/malloryprow/EVS/blob/502f33d3e594e38845c6d37ea3200aa2be24dddd/parm/evs_config/subseasonal/config.evs.subseasonal.grid2grid.plots.sea_ice#L49)

PerryShafran-NOAA commented:
Hi, @ShannonShields-NOAA - two other items that need addressing.  I've been told that you need to remove several items from your driver scripts.  Those items are:

COMINobs
FIXevs
COMINclimo

These all should be taken care of in your J-jobs (and they are).  You will also need to make a symbolic link to fix in your HOMEevs directory.  You might need to run a test for this; or I could run a test here and see what happens.  The removal of these items shouldn't change the execution of the code.  

These changes might require a test of the code, but probably just the early part to see that all these items are set correctly.

Perry

PerryShafran-NOAA commented:
@AliciaBentley-NOAA @ShannonShields-NOAA Just noticed that the www path seems to look like the emcrzdm directory where the output plot file would reside?  I think that there should be nothing in EVS that refers to emcrzdm, so the SEND2WEB block of code here should be removed in its entirety.  Would there be any other place in the subseasonal component that refers to sending data to emcrzdm?  

Perry

AliciaBentley-NOAA commented:
@ShannonShields-NOAA In case you're curious, all of the comments above are coming from the [subseasonal section of the EE2 checklist](https://docs.google.com/document/d/1fYsVrcrjj_NxJBYG6v0sDhi-vYxShT2l2EUauHIByLs/edit?pli=1) that I've been compiling. I mentioned it in Slack earlier today (Monday). 

@PerryShafran-NOAA I agree that there shouldn't be anything in the ops code delivery that refers to "www" or "emcrzdm." I would be comfortable with Shannon removing more than just one line to clean up this section if doing so doesn't break the subseasonal component.

AliciaBentley-NOAA commented:
@ShannonShields-NOAA Continuing to add to the EE2 checklist document, but adding here as well:

- Remove lines 203–215 (that are commented out, but refer to “www” path): [ush/subseasonal/set_up_subseasonal_evs.sh](https://github.com/NOAA-EMC/EVS/blob/077ab1f895d77091b60636026d7b3b6c5e81cade/ush/subseasonal/set_up_subseasonal_evs.sh#L203-L215)

ShannonShields-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/subseasonal_DCOMROOT_removal

ShannonShields-NOAA commented:
Sorry all, I am trying to get caught up with everything. So yes, the rzdm
"www" that you are referring to in my code is old and outdated; I was going
to go through my code this week and remove any old, unused lines of code to
submit in a feature/subseasonal_finalization PR. But I can do this in this
PR.
Perry, I'm afraid I don't know exactly what you mean with the COMINobs,
FIXevs, COMINclimo paths removal. I'm not sure if the code will work if
these are removed. Can you test what you mean in the parallel or something?
Thanks,
Shannon

On Tue, Sep 12, 2023 at 9:19 AM Alicia Bentley ***@***.***>
wrote:

> @ShannonShields-NOAA <https://github.com/ShannonShields-NOAA> Continuing
> to add to the EE2 checklist document, but adding here as well:
>
>    - Remove lines 203–215 (that are commented out, but refer to “www”
>    path): ush/subseasonal/set_up_subseasonal_evs.sh
>    <https://github.com/NOAA-EMC/EVS/blob/077ab1f895d77091b60636026d7b3b6c5e81cade/ush/subseasonal/set_up_subseasonal_evs.sh#L203-L215>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/231#issuecomment-1715712047>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP4MEUGZN7BGKCDYAPDRHWLX2BONRANCNFSM6AAAAAA4PJ3WYI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Shannon Shields
SAIC at NOAA/NWS/NCEP/EMC
5830 University Research Ct. Rm 2180
College Park, MD 20740
***@***.***

AliciaBentley-NOAA commented:
@ShannonShields-NOAA It's ok! It might be a little overwhelming to see all of this in writing, but it's actually super easy. These edits are really small. [Please click on this link to the EE2 checklist](https://docs.google.com/document/d/1fYsVrcrjj_NxJBYG6v0sDhi-vYxShT2l2EUauHIByLs/edit?pli=1#heading=h.q91us2dv22u4) and navigate to subseasonal. Please make all of these edits, which are mostly one line changes. 

As for the part about removing COMINobs etc., from your /dev/driver scripts, the key is that they should actually be defined in our J-jobs. So they'll still be defined, just in your J-jobs. This will clean up and streamline Lin Gan's ecflow scripts. Here is what is written in the EE2 checklist so that this is clearer. I highly recommend that you refer to this document (linked above). 

- Remove export COMINobs, export FIXevs, and export COMINclimo (L58–60) in [dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_gefs_grid2obs_stats.sh](https://github.com/NOAA-EMC/EVS/blob/de168d6d4e6a9ed157eb350b914b906aaada9479/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_gefs_grid2obs_stats.sh#L58-L60) **and make sure they are defined in J-jobs instead**
- Remove export COMINobs, export FIXevs, and export COMINclimo (L58–60) in [dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh](https://github.com/NOAA-EMC/EVS/blob/de168d6d4e6a9ed157eb350b914b906aaada9479/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh#L58-L60) **and make sure they are defined in J-jobs instead**

ShannonShields-NOAA commented:
Ok, thank you Alicia; will do.

On Tue, Sep 12, 2023 at 10:34 AM Alicia Bentley ***@***.***>
wrote:

> @ShannonShields-NOAA <https://github.com/ShannonShields-NOAA> It's ok! It
> might be a little overwhelming to see all of this in writing, but it's
> actually super easy. These edits are really small. Please click on this
> link to the EE2 checklist
> <https://docs.google.com/document/d/1fYsVrcrjj_NxJBYG6v0sDhi-vYxShT2l2EUauHIByLs/edit?pli=1#heading=h.q91us2dv22u4>
> and navigate to subseasonal. Please make all of these edits, which are
> mostly one line changes.
>
> As for the part about removing COMINobs etc., from your /dev/driver
> scripts, the key is that they should actually be defined in our J-jobs. So
> they'll still be defined, just in your J-jobs. This will clean up and
> streamline Lin Gan's ecflow scripts. Here is what is written in the EE2
> checklist so that this is clearer. I highly recommend that you refer to
> this document (linked above).
>
>    - Remove export COMINobs, export FIXevs, and export COMINclimo
>    (L58–60) in
>    dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_gefs_grid2obs_stats.sh
>    <https://github.com/NOAA-EMC/EVS/blob/de168d6d4e6a9ed157eb350b914b906aaada9479/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_gefs_grid2obs_stats.sh#L58-L60> *and
>    make sure they are defined in J-jobs instead*
>    - Remove export COMINobs, export FIXevs, and export COMINclimo
>    (L58–60) in
>    dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh
>    <https://github.com/NOAA-EMC/EVS/blob/de168d6d4e6a9ed157eb350b914b906aaada9479/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh#L58-L60> *and
>    make sure they are defined in J-jobs instead*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/231#issuecomment-1715848104>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP4MEUGWUCJYGLSDATLXNNLX2BXHVANCNFSM6AAAAAA4PJ3WYI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Shannon Shields
SAIC at NOAA/NWS/NCEP/EMC
5830 University Research Ct. Rm 2180
College Park, MD 20740
***@***.***

Commit:
Use  for all /prod/dcom paths, remove www paths, delete old scripts

ShannonShields-NOAA commented:
@AliciaBentley-NOAA and @PerryShafran-NOAA I have committed most of the changes from the EE2 checklist to this PR. I changed to the prod/dcom path, I removed the old "SEND2WEB" section which had www path in configs, I deleted some old scripts, and I changed the maillist.

ShannonShields-NOAA commented:
However, I am still confused on some of the edits concerning the FIX directory. Under my feature branch, /lfs/h2/emc/vpppg/noscrub/shannon.shields/feature_subseasonal_DCOMROOT_removal/EVS, I need to mkdir fix?
The thing is HOMEevs is "export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS" so this will not work in a feature branch name. It seems to me that this link needs to be done in the auth repo by Perry or Shelley, especially since it exists outside of any components.

AliciaBentley-NOAA commented:
@ShannonShields-NOAA  Lin Gan required FIXevs to be removed from our /dev/drivers and defined in our J-jobs only as: 
export FIXevs=${FIXevs:-$HOMEevs/fix}

In order for you to delete FIXevs from your /dev/drivers and also continue to run your own cronjobs using this code, you need a way to tell your code that the fix directory is located at $HOMEevs/fix. The symbolic link allows you to do that without actually copying the entire /EVS_fix directory into your $HOMEevs directory. Does that makes sense? I think that our directions will work for your needs. They've already worked for others, if that helps!

- cd into your HOMEevs directory
- Create a new /fix directory using mkdir fix
- Set up a symbolic link between the new /fix and the actual /EVS_fix directory:
ln -s /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix/* fix/

PerryShafran-NOAA commented:
Hi, @ShannonShields-NOAA ,

Generally what we (myself or Shelley) would do is in our testing we would do the following:

ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix

Before running anything.  That ensure that FIXevs=$HOMEevs/fix (as defined in the J-job) will find all the items on the fix directory that you need.  

However, you too should do this in your own online test as well to ensure that this works.  It has worked well in other tests, but the key for you is to ensure that this definition of FIXevs is defined in your J-job script.  However, you should not post fix in your fork (nor will it wind up in the repository - but it *is* in the emc.vpppg parallel).  We will do it ourselves *and* you need to do it in your own HOMEevs directory.

Perry

ShannonShields-NOAA commented:
Ok, so I think I am understanding better now...so in my current working directory on WCOSS2, I want to remove FIX export and mkdir fix. However, in the feature branch on GitHub I just want to remove the "export FIXevs=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix" in my dev'driver script and NOT actually make a fix directory that will show up alongside the "dev", "jobs", modulefiles etc..

AliciaBentley-NOAA commented:
@ShannonShields-NOAA You want to remove "export FIXevs=...." from all of your /dev/driver scripts. You then want define FIXevs in your J-jobs as: export FIXevs=${FIXevs:-$HOMEevs/fix}

I don't think you want to mkdir fix in your **working** directory. You said that your working directory is /lfs/h2/emc/vpppg/noscrub/shannon.shields/feature_subseasonal_DCOMROOT_removal/EVS, I believe. You should "mkdir fix" in whatever your $HOMEevs directory is. This will allow the definition of FIXevs in your J-jobs to find $HOMEevs/fix. Thanks.

PerryShafran-NOAA commented:
Here is what you need to do:

1) You do need a fix directory that is symbolically linked to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix.  Like this:
ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix
2) You must *not* commit this directory to github though.  
3) You must remove any export FIXevs statements from all your dev/drivers/scripts files.  
4) You must ensure that FIXevs is defined as $HOMEevs/fix is defined in your J-jobs (exactly as Alicia defines above).

I think that this nicely summarizes what you need to do and what you shouldn't do.

Commit:
Remove export FIXevs in dev/drivers/scripts

ShannonShields-NOAA commented:
Ok, thank you both for explaining! I finally understand what you mean and made the commit as you can see.

ShannonShields-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/subseasonal_DCOMROOT_removal

ShannonShields-NOAA commented:
@PerryShafran-NOAA Now that we have the machine back, this PR is ready to test.
First, please run a test for the obs prep job (mainly testing the new DCOMROOT path) via EVS/dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_obs_prep.sh.
This should be a quick test; make sure you set HOMEevs path correctly to match the feature branch.

PerryShafran-NOAA commented:
Will do!

Perry

PerryShafran-NOAA commented:
BTW, you still have FIXevs in this driver script:

export FIXevs=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix

in the jevs_subseasonal_obs_prep.sh.

I will remove this line for the test, but you will also need to remove it from your driver script.

Perry

ShannonShields-NOAA commented:
Oh right! The EE2 doc only listed the stats files, but I forgot this is also in prep (and maybe plots) so I will go through and make sure this line is removed from all dev/drivers scripts. This also means that you will likely need to do that link to HOMEevs/fix for this to work.

Commit:
Remove export FIXevs from dev/drivers/scripts

ShannonShields-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/subseasonal_DCOMROOT_removal

AliciaBentley-NOAA commented:
@ShannonShields-NOAA Looks great! Thanks! Question for you: Are we treating this PR as your clean-up PR now (meaning your won't be submitting another cleanup PR)? If so, then there are additional EE2 edits that you can make to all of your /dev/driver scripts in this PR. 

The main thing is to remove as many "export COMIN____" lines as possible from you /dev/driver scripts and make sure that they are defined in your J-jobs instead. Examples of lines that consistently have the same definition and can therefore be moved to J-jobs could include [some of these "export COMIN_____" lines ](https://github.com/NOAA-EMC/EVS/pull/231/files#diff-6a9dfc072fde7cea18e4466aa5bf3d72c0d5a75f42471e4c2a3566c2ce4ebb72L53-L60). Please check all /dev/drivers. 

Apologies that I didn't write out every script and line explicitly for you in the EE2 document. When I do searches of the EVS repository on github for issues, it only gives me 5 pages of matches to my search. There are so many COMIN edits to make to the auth repo that the other subseasonal /dev/drivers weren't included in the first 5 pages of search results. Thanks for understanding. We're all trying our best.

ShannonShields-NOAA commented:
Of course @AliciaBentley-NOAA , no worries; these are trying times for everyone. 
While I was not treating this as my final clean-up PR (that was going to be feature/subseasonal_finalization), I can commit those COMIN removal changes to this PR while it is still open. So just to clarify, delete all instances of "export COMIN***" in dev/drivers, but leave in "export COMOUT" as is? Or should COMOUT be removed too?

AliciaBentley-NOAA commented:
@ShannonShields-NOAA Lines that can stay in your /dev/drivers/scripts are lines that need to be defined differently than they are in your j-jobs. For example, in a global_det prep driver, the only thing left is [export COMOUT](https://github.com/NOAA-EMC/EVS/blob/develop/dev/drivers/scripts/global_det/prep/jevs_global_det_atmos_prep.sh#L41) because it may need to point to a special "dev" location. Just make sure that each definition ends in the same directory in your /dev/drivers and J-jobs. For example, from global_det, both end with $RUN, which is good. 


dev/driver definition:
export COMOUT=/lfs/h2/emc/vpppg/noscrub/$USER/$NET/$evs_ver/$STEP/$COMPONENT/$RUN
J-job definition: 
export COMOUT=${COMOUT:-$(compath.py -o $NET/$evs_ver/$STEP/$COMPONENT/$RUN)}

Commit:
Remove export COMIN from dev/drivers/scripts

AliciaBentley-NOAA commented:
@ShannonShields-NOAA Thanks for making these changes. You may not know this yet, but it's actually required by NCO that we use compath.py to make all of the COMIN and COMOUT directory paths in your J-jobs. I see that you have stopped using it in your recent edits. Can you please bring it back? You might have to modify it slightly. You can use compath.py on the command line to make sure it's showing you what you need it to. 

Was:
export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver})}
You set it to:
export COMIN=${COMIN:-$COMROOT/$NET/$evs_ver/prep/$COMPONENT/$RUN}
Can it be tested as:
export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver}/prep/$COMPONENT/$RUN)}

Thanks!

Commit:
Fix COMIN and COMOUT in dev/drivers/scripts

ShannonShields-NOAA commented:
@AliciaBentley-NOAA Sorry, I have fixed that in the most recent commit.
@PerryShafran-NOAA Phew, ok, that was a lot of changes. Please retest the obs prep job by running EVS/dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_obs_prep.sh. Remember I didn't make the commit for the fix directory so you have to mkdir fix in HOMEevs and link to FIXevs!

PerryShafran-NOAA commented:
OK, the jevs_subseasonal_obs_prep.sh run is complete.  

The .o file is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr231test/EVS/dev/drivers/scripts/subseasonal/prep
The output is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos.20230911

Please have a look and let me know if it looks good.

Perry

ShannonShields-NOAA commented:
Everything looks good and ran successfully with the obs prep job. 
Next, please test the stats job. First, copy over prep data from me (/lfs/h2/emc/vpppg/noscrub/shannon.shields/EVS_Data/evs/v1.0/prep/subseasonal/atmos.20230807 to the present), and then run the following:
 EVS/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh

PerryShafran-NOAA commented on dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh:
```diff
@@ -29,7 +29,6 @@ source $HOMEevs/modulefiles/subseasonal/subseasonal_stats.sh
 export MET_ROOT=/apps/ops/para/libs/intel/${intel_ver}/met/${met_ver}
 export MET_BASE=${MET_ROOT}/share/met
 export PATH=${MET_ROOT}/bin:${PATH}
```
I think that these three lines should probably deleted.  The modulefile takes care of bringing in the current version of MET, and plus you're pointing to para here.  Are you using MET_ROOT and MET_BASE throughout your code?

ShannonShields-NOAA replied:
I do use MET_ROOT and METPLUS_PATH as environ variables throughout all my ush scripts. I'm not sure why it still says para, but I can change it to the prod path.

AliciaBentley-NOAA replied:
@ShannonShields-NOAA Nothing in EVS should says para for the ops code delivery, so change this to prod, keep it all for now, and we'll worry about it later as a  bugfix or EE2 fix. 

Based on the [loading instructions for MET-11.0.2 in /prod](https://dtcenter.org/community-code/metplus/metplus-5-0-existing-builds), this line might be slightly changed to:
module use /apps/prod/lmodules/intel/19.1.3.304

ShannonShields-NOAA replied:
Also, the parm/machine.conf uses MET_INSTALL_DIR = {ENV[MET_ROOT]}. So I think the scripts will fail if these lines are removed.

PerryShafran-NOAA replied:
I think that MET_ROOT is included as part of the met library install.  I do not have MET_ROOT defined in any of my components and MET still works.  

In the working test below (with the COMIN fix), I actually commented out these lines:

```
#export MET_ROOT=/apps/ops/para/libs/intel/${intel_ver}/met/${met_ver}
#export MET_BASE=${MET_ROOT}/share/met
#export PATH=${MET_ROOT}/bin:${PATH}
```

So I feel you can safely remove all of these lines.

Perry

PerryShafran-NOAA commented:
@ShannonShields-NOAA Unfortunately, you have COMIN and COMINobs undefined:

```
 + export COMIN=
1 + COMIN=
1 + export COMINobs=
1 + COMINobs=
```

I see that @AliciaBentley-NOAA has asked you to remove COMIN from the driver script?  I feel that you should put that back in for the time being.  I think that this would cause fewer problems, considering that the COMIN data is not coming in from any production directory.  The setting:

`export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver}/prep/$COMPONENT/$RUN)}`

does not work.  

I can also defer to @AliciaBentley-NOAA for suggestions as well.  

The prep directory, for reference is /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos.20230911.  There's no $envir or com in that path.

AliciaBentley-NOAA commented:
@PerryShafran-NOAA Are you working on the cfs stats script now? What I defined would have probably worked for prep, but not stats. 

In stats J-jobs, other people have COMIN defined this way:
export COMIN=${COMIN:-$(compath.py -o $NET/$evs_ver)}

I think Shannon needs to change COMIN in her stats J-job to:
export COMIN=${COMIN:-$(compath.py -o $NET/$evs_ver/prep/$COMPONENT/$RUN)}

For now, you can put it back in the driver, but Lin Gan is going to run it without the dev drivers and we'll have to fix things then anyway! Thanks!

PerryShafran-NOAA commented:
@ShannonShields-NOAA Let's try Alicia's latest fix of COMIN.  That just might work.  I can test it out.

PerryShafran-NOAA commented:
I have confirmed that the directory with this latest fix gives us this:

```
1 + export COMIN=/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos
1 + COMIN=/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos
1 + export COMINobs=/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos
1 + COMINobs=/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos
```

The output file is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr231test/EVS/dev/drivers/scripts/subseasonal/stats 
The stats directory is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/subseasonal/*20230911*

Perry

AliciaBentley-NOAA commented:
@ShannonShields-NOAA Please change the path to COMIN in your stats J-job to:
export COMIN=${COMIN:-$(compath.py -o $NET/$evs_ver/prep/$COMPONENT/$RUN)}

Is that correct, @PerryShafran-NOAA ? Did you just copy my suggested COMIN line from above?

PerryShafran-NOAA commented:
@AliciaBentley-NOAA - yep, I copied your suggestion into the J-job and it worked.

Commit:
Fix COMIN and MET_ROOT for testing

ShannonShields-NOAA commented:
As you can see in my latest commit, I made the COMIN change and I set MET_ROOT=
/apps/prod/lmodules/intel/${intel_ver}/met/${met_ver}.
Everything is good with the stats job!
If you have time, @PerryShafran-NOAA please make sure you have the latest changes and then run:
EVS/dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2grid_anom_plots_31days.sh

AliciaBentley-NOAA commented:
@ShannonShields-NOAA I had a funny feeling about the path that I got from the DTC page. I think the "module use" path is an alias instead of an actual directory, so I went and checked if 11.0.2 was in there... and it's not! Instead, it is at the /prod version of the old /para path: /apps/ops/**prod**/libs/intel/19.1.3.304/met/11.0.2

Sorry for leading you astray there! It was the EVS VE that was moved to a whole new path, not MET! 

Before you had:
export MET_ROOT=/apps/ops/para/libs/intel/${intel_ver}/met/${met_ver}
Which should be:
export MET_ROOT=/apps/ops/**prod**/libs/intel/${intel_ver}/met/${met_ver}

Thanks for making this change!

Commit:
Fix MET_ROOT prod path

ShannonShields-NOAA commented:
@AliciaBentley-NOAA Ah yes, I was thinking it needed to be changed to that module path; thank you for catching that!
@PerryShafran-NOAA Ok, now this PR should have all the changes it should need. Please make sure to recheck out this feature branch with the updated edits and run:
EVS/dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2grid_anom_plots_31days.sh

PerryShafran-NOAA commented:
I think you're in business.  Finally got a successful plot tar file.  

Check out the .o file /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr231test/EVS/dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2grid_anom_plots_31days.o81856958

and the output tar file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/plots/subseasonal/atmos.20230912

Perry

ShannonShields-NOAA commented:
Yes! All looks good to me.
Phew, sorry, I know that was a lot of back and forth with all the edits; I appreciate the help.
This PR should finally be good to go to merge.

PerryShafran-NOAA commented:
OK, will do one last quickie code review, and then if all looks good, merge into develop.

PerryShafran-NOAA commented:
Approved for merge.
